sympy/core/numbers.py

```diff
@@ -211,6 +211,7 @@ def igcd(*args):
 except ImportError:
     def igcd2(a, b):
         """Compute gcd of two Python integers a and b."""
+        a, b = as_int(a), as_int(b)
```

Comment:
It is coming from [here](https://github.com/sympy/sympy/blob/5522219676a18f250582ee39654799d80ef40c24/sympy/polys/ring_series.py#L975). Apparently `PythonRational.__pow__` uses [`self.new`](https://github.com/sympy/sympy/blob/5522219676a18f250582ee39654799d80ef40c24/sympy/polys/domains/pythonrational.py#L201), which is probably the original way of avoiding gcd computation. 

In that code, is `c` always supposed to be an `n`th power (is that what the `DomainError` catches?)? If so, we can replace it with `integer_nthroot`.